drivers/thermal/k1x-thermal.c

```diff
@@ -286,14 +287,23 @@ static int k1x_thermal_probe(struct platform_device *pdev)
 	for (i = s->sr[0]; i <= s->sr[1]; ++i) {
 		s->sdesc[i].base = s->base;
 
-		s->sdesc[i].tzd = devm_thermal_of_zone_register(dev,
-				i, s->sdesc + i, &k1x_of_thermal_ops);
-		if (IS_ERR(s->sdesc[i].tzd)) {
+		struct thermal_zone_device *tzd =
+			devm_thermal_of_zone_register(dev, i, s->sdesc + i, &k1x_of_thermal_ops);
+		if (IS_ERR(tzd)) {
 			ret = PTR_ERR(s->sdesc[i].tzd);
```

Comment:
should update 'ret' too, like ret = PTR_ERR(tzd)
also it would be great to use dev_err_probe() to combine with ret & dev_err()